packages/cloudflare/test/fixtures/dev-runtime/astro.config.ts

```diff
@@ -5,6 +5,19 @@ export default defineConfig({
 	adapter: cloudflare({
 		runtime: {
 			mode: 'local',
+			type: 'pages',
+			bindings: {
+				bindings: {
+					"COOL": "ME"
+				},
```

Comment:
The double `bindings` feel weird. I am sure there's a reason for that.

Reply:
Hey @dario-piotrowicz, could you share your thoughts with an explanation? Your insights are always the best we can get, to make sure we understand Cloudlare's side!

Also, what's your take on an alternative approach to avoid the `bindings.bindings` scenario? Looking forward to your ideas!

Reply:
Sure, thanks for pinging me 🙂 

I totally agree that `bindings.bindings` is quite awkward, it is just that this closely follows [Miniflare's API](https://github.com/cloudflare/workers-sdk/tree/main/packages/miniflare#core) (which has an unfortunate/confusing name for this property), so having this `bindings.bindings` structure is correct although awkward.

As alternatives there are two that I can think of, which basically involve diverging from the Miniflare's format:
 - you could rename the `bindings` field inside `bindings` something like `textBindings` or `jsonBindings` (that's what I did a couple of weeks ago [in our Next.js adapter](https://github.com/cloudflare/next-on-pages/blob/c9167ea4bd11cb235e1d0545f25f187fd83771fb/internal-packages/next-dev/src/index.ts#L31), but I am pretty sure I am going to revert that and going with the next alternative soon👇). This would allow you to keep things almost in sync with the Miniflare API and diverge minimally from it.
 - you could restructure the whole `bindings` object to follow a structure like `bindings?: Record<string, Binding>` (that I just learned that it's getting introduced [in wrangler](https://github.com/cloudflare/workers-sdk/blob/eaa16db25103d26ef31499634a31b86caccdf7aa/packages/wrangler/src/api/startDevWorker/types.ts#L34)) (in that case these text bindings would practically be [`var` type bindings](https://github.com/cloudflare/workers-sdk/blob/eaa16db25103d26ef31499634a31b86caccdf7aa/packages/wrangler/src/api/startDevWorker/types.ts#L216C1-L216C1)). This to me feels like the better approach but it would require you to take the user options and mapping them to the current Miniflare API. Also I am not sure how stable these APIs are in wrangler as it is current work in progress I believe. (PS: I _think_ that a future Miniflare version could use a structure similar to this as well)
 
As I said I think that the second alternative is better (sorry for not mentioning this any earlier, I actually had something like this in mind but I was totally unaware that the wrangler team already defined it there! 😓).
 
Please let me know what you think 🙂

Reply:
Thank you for the insights; they help a lot.

Personally, I think it would be best to go with the second approach. Even though it might not be 100% stable and Cloudflare could change it again, I believe it is the more robust API decision. This approach offers us the most flexibility while still keeping us in sync with Cloudflare's design.

What do you think, @ematipico? Would this approach align with your preferences?

Reply:
Sounds good to me too!